File: 8kyu/CSV-representation-of-array.go
/* 8 kyu
CSV representation of array

https://www.codewars.com/kata/5a34af40e1ce0eb1f5000036/go

Create a function that returns the CSV representation of a two-dimensional numeric array.

Example:

input:
   [[ 0, 1, 2, 3, 4 ],
    [ 10,11,12,13,14 ],
    [ 20,21,22,23,24 ],
    [ 30,31,32,33,34 ]]

output:
     '0,1,2,3,4\n'
    +'10,11,12,13,14\n'
    +'20,21,22,23,24\n'
    +'30,31,32,33,34'
Array's length > 2.
*/

// Helped by chatGPT

package kata

import (
	"strconv"
	"strings"
)

func ToCsvText(array [][]int) string {
	var result strings.Builder

	for i, row := range array {
		for j, value := range row {
			result.WriteString(strconv.Itoa(value))

			if j < len(row)-1 {
				result.WriteRune(',')
			}
		}

		if i < len(array)-1 {
			result.WriteRune('\n')
		}
	}

	return result.String()
}

File: 8kyu/CSV representation of array.ts
/**
 * 8 kyu
 * CSV representation of array
 * https://www.codewars.com/kata/5a34af40e1ce0eb1f5000036/typescript
 *
 * Converts a two-dimensional number array into a CSV-style string,
 * joining each inner array with commas and separating rows with newline characters ("\n").
 *
 * @param array - A two-dimensional array of numbers.
 * @returns A string where each row of numbers is separated by a newline.
 *
 * @example
 * toCsvText([[ -25, 21, 2, -33, 48 ], [ 30, 31, -32, 33, -34 ]]);
 * // '-25,21,2,-33,48\n30,31,-32,33,-34'
 */
export function toCsvText(array: number[][]): string {
  return array.join("\n");
}
